ui/cap-react/src/antd/forms/fields/DataDbField.js

```diff
@@ -0,0 +1,146 @@
+  const [showModal, setShowModal] = useState(
+    !formData || (formData && Object.keys(formData).length == 0)
+  );
+  const [ldapData, setLdapData] = useState([]);
+  const [tableLoading, setTableLoading] = useState(false);
+  const [form] = Form.useForm();
+
+  const modalColumns = [
+    {
+      title: "Name",
+      dataIndex: "name",
+      key: "name"
+    },
+    {
+      title: "Email",
+      dataIndex: "email",
+      key: "email"
+    },
+    {
+      title: "Department",
+      dataIndex: "department",
+      key: "department",
+      render: tag => <Tag color="geekblue">{tag}</Tag>
+    },
+    {
+      title: "Action",
+      key: "action",
+      render: item => (
+        <Button
+          type="primary"
+          onClick={() => {
+            setShowModal(false);
+            form.resetFields();
+            setLdapData([]);
+            onChange({
+              name: item.name,
+              email: item.email,
+              department: item.department
+            });
+          }}
+        >
+          Add
+        </Button>
+      )
+    }
+  ];
+
+  const fetchLDAPdata = _debounce(async ({ searchFor, searchInput }) => {
+    setTableLoading(true);
+    const response = await axios.get(
+      `/api/services/ldap/${searchFor}/mail?query=${searchInput}`
+    );
+
+    setLdapData(
+      response.data.map(item => ({
+        target: searchFor,
+        key: item.email ? item.email : item,
+        email: item.email ? item.email : item,
+        department: item.email ? item.profile.department : "egroup",
+        name: item.email ? item.profile.display_name : item.split("@cern.ch")[0]
+      }))
+    );
+    setTableLoading(false);
+  }, 500);
+
+  return (
+    <React.Fragment>
+      <Modal
+        visible={showModal}
+        onCancel={() => {
+          setShowModal(false);
+          form.resetFields();
+          setLdapData([]);
+        }}
+        width={1000}
+      >
+        <Form
+          form={form}
+          initialValues={{ searchFor: "user" }}
+          layout="vertical"
+          onValuesChange={(_, values) =>
+            values.searchInput && values.searchInput.length > 0
+              ? fetchLDAPdata(values)
+              : setLdapData([]) && setTableLoading(false)
+          }
+        >
+          <Form.Item label="Search For" name="searchFor">
+            <Radio.Group buttonStyle="solid">
+              <Radio.Button value="user">Users</Radio.Button>
+              <Radio.Button value="egroup">E groups</Radio.Button>
+            </Radio.Group>
+          </Form.Item>
+          <Form.Item name="searchInput">
+            <Input placeholder="Search for emails..." />
+          </Form.Item>
+        </Form>
+        <Table
+          loading={tableLoading}
+          dataSource={ldapData}
+          columns={modalColumns}
+          fixedHeader
+          pagination={false}
+        />
+      </Modal>
+
+      {Object.keys(formData).length > 0 ? (
```

Comment:
initially when `null/undefined` this will break..maybe just check if `formData` is there